shell/config/router/navigation-guards/attempt-first-login.js

```diff
@@ -0,0 +1,71 @@
+import { SETUP } from '@shell/config/query-params';
+import { SETTING } from '@shell/config/settings';
+import { MANAGEMENT, NORMAN } from '@shell/config/types';
+import { tryInitialSetup } from '@shell/utils/auth';
+import { routeNameMatched } from '@shell/utils/router';
+
+export function install(router, context) {
+  router.beforeEach((from, to, next) => attemptFirstLogin(from, to, next, context));
+}
+
+export async function attemptFirstLogin(from, to, next, { store }) {
+  if (routeNameMatched(to, 'unauthenticated')) {
+    return next();
+  }
```

Comment:
Ensures we only run when in a route that uses the 'authenticated' parent route.

I have a feeling we're going to want to move this to a mixin at some point. As of right now running this code in a route beforeEach() is the closest in the execution timeline to middleware.